src/components/number/number.component.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useRef } from "react";
+import Textbox, { TextboxProps } from "../textbox";
+
+export interface NumberProps extends Omit<TextboxProps, "value"> {
+  /** Value passed to the input */
+  value?: string;
+}
+
+function isValidNumber(value: string) {
+  const regex = new RegExp("^[-]?[0-9]*$");
+  const result = regex.test(value);
+
+  return result;
+}
+
+export const Number = ({
+  onChange,
+  onKeyDown,
+  value,
+  ...rest
+}: NumberProps) => {
+  const selectionStart = useRef<null | number>(null);
+  const selectionEnd = useRef<null | number>(null);
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (isValidNumber(event.target.value) && onChange) {
+      onChange(event);
+    } else {
+      event.target.value = value || "";
+      event.target.setSelectionRange(
+        selectionStart.current,
+        selectionEnd.current
+      );
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    selectionStart.current = (event.target as HTMLInputElement).selectionStart;
+    selectionEnd.current = (event.target as HTMLInputElement).selectionEnd;
+
+    if (onKeyDown) {
+      onKeyDown(event);
+    }
+  };
+  return (
+    <Textbox
+      {...rest}
+      value={value}
+      onChange={handleOnChange}
+      onKeyDown={handleKeyDown}
+    />
+  );
+};
+
+Number.displayName = "Number";
```

Comment:
do we really have to explicitly set the displayName here? I thought React would set that automatically, unless we were using forwardRef or something (which we're not, at least for now).

It won't do any harm (and may save us some work/bugs in future), I'm just surprised it's been added as part of a TS refactor.

Reply:
I'm pretty sure that the `Number` component should be wrapped in a `forwardRef`. I'll remove it.